api/v2/api.js

```diff
@@ -134,6 +142,24 @@ router.get('/random', async (req, res) => {
 
 });
 
+router.get('/grumble', async(req, res) => {
+
+  let limit = parseInt(req.query.lat) || 10;
+  if (isNaN(limit)) { return res.status(400).send('Invalid Parameter: LIMIT'); }
```

Comment:
I think we can account for more edge cases by checking if limit is an integer or not (rather than Nan)

Also, I personally prefer L147 and L148 to be moved down right before the for loop on L157 (just so logic is more closely grouped). But this is a personal preference, so no need to do son

Reply:
L147 returns either the result from parseInt() or 10 which is already an integer. parseInt() only returns integer or NaN depending on if it can parse the string passed to it, so no need to check for int

Reply:
And L147/148 is up here so that if an invalid input is passed through the limit parameter, the api doesn't spend the resources and time to call the google api and return results

Reply:
oh, I didn't even realize the parseInt() call

ok yea, having it up here makes sense. You're right